datahub/opal.py

```diff
@@ -140,15 +96,23 @@ def get_opal_row(data: dict[str, float]) -> pd.Series:  # type: ignore[type-arg]
     Returns:
         A pandas Series containing the new data
     """
-    data_array = []
+    if "array" not in data:
+        data_index = data["frame"]
+
+        data_array = []
+
+        for item in opal_headers.values():
+            if item in data:
+                data_array.append(data[item])
```

Comment:
This should be able to be condensed into a [list comprehension](https://realpython.com/list-comprehension-python/)